source/extensions/filters/http/grpc_field_extraction/message_converter/message_converter.cc

```diff
@@ -0,0 +1,159 @@
+#include "source/extensions/filters/http/grpc_field_extraction/message_converter/message_converter.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+
+#include "absl/log/absl_check.h"
+#include "grpc_transcoding/message_reader.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcFieldExtraction {
+
+namespace {
+using Envoy::Protobuf::io::ZeroCopyInputStream;
+} // namespace
+
+MessageConverter::MessageConverter(std::unique_ptr<CreateMessageDataFunc> factory,
+                                   uint32_t buffer_limit)
+    : factory_(std::move(factory)), buffer_limit_(buffer_limit),
+      parsed_bytes_usage_(std::make_shared<uint64_t>(0)) {}
+
+absl::StatusOr<std::unique_ptr<StreamMessage>>
```

Comment:
style nit: define a type alias for StreamMessagePtr?

Reply:
Done.